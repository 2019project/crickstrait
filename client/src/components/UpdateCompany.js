import React, { Component } from "react";
import { connect } from "react-redux";
import "../styles/Admin.css";

import { updateCompany } from "../actions/UpdateCompany";
import SampleTable from "./Common/SampleData.png";

export class UpdateCompany extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedFile: null
    };
  }

  onChangeHandler = event => {
    this.setState({
      selectedFile: event.target.files[0]
    });
  };
  onClickHandler = () => {
    const data = new FormData();
    data.append("file", this.state.selectedFile);

    this.props.updateCompany(data, this.props.id);
    // console.log(this.state);
  };

  render() {
    // console.log(this.props);
    return (
      <div id="updateCompanyMainDiv">
        <a className="admin_close" href="#stocks_main_grid_container">
          &times;
        </a>
        {this.state.selectedFile != null ? (
          <p></p>
        ) : (
          <>
            <p id="updateCompanyMainDivText1">Select a File</p>
            <div id="updateCompany_tooltip_div">
              <i class="fa fa-info-circle" />
              <span id="updateCompany_tooltip_text">
                <ul className="updateCompany_tooltip_text_list">
                  <li>
                    File must be a <b>csv</b> file.
                  </li>
                  <li>
                    <b>Indicator Names</b> must be the same as used in the
                    Application.
                  </li>
                  <li>
                    Indicator Names on <b>left</b> and their Data on{" "}
                    <b>right</b>
                  </li>
                  <li>
                    Date for the data must be in ISO format
                    <b>("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")</b>.
                  </li>
                  <li>It should follow the below pattern.</li>
                </ul>
                <img
                  src={SampleTable}
                  alt="Sample Table"
                  className="updateCompany_table_image"
                />
              </span>
            </div>
          </>
        )}
        <input
          className="updateCompany_admin_input"
          style={
            this.state.selectedFile != null
              ? { padding: "12px 35px 0px 25px" }
              : { padding: "12px 35px 0px 35px" }
          }
          type="file"
          name="file"
          onChange={this.onChangeHandler}
        />

        <button
          type="button"
          className="btn btn-success btn-block updateCompany_admin_button"
          onClick={this.onClickHandler}
        >
          <i className="fa fa-upload" style={{ marginRight: "5px" }}></i>
          <a
            href="#stocks_main_grid_container"
            style={{ textDecoration: "none" }}
          >
            Upload
          </a>
        </button>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  //  isLoading: state.LoadingReducer.isLoading
});

export default connect(mapStateToProps, { updateCompany })(UpdateCompany);
