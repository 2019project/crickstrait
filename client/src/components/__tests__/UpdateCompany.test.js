import React from "react";
import { UpdateCompany } from "../UpdateCompany";
import { shallow } from "enzyme";

const updateCompanyData = jest.fn();
const updateCompany = jest.fn();

const wrapper = shallow(
  <UpdateCompany
    updateCompanyData={updateCompanyData}
    updateCompany={updateCompany}
  />
);

describe("Testing Navbar Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a division with name updateCompanyMainDiv", () => {
    expect(wrapper.find("#updateCompanyMainDiv")).toBeTruthy();
  });
  it("should have a division with name updateCompany_tooltip_div", () => {
    expect(wrapper.find("#updateCompany_tooltip_div")).toBeTruthy();
  });
  it("should display the Select a File", () => {
    expect(wrapper.find("#updateCompanyMainDivText1").text()).toBe(
      "Select a File"
    );
  });

  it("should have null in state at the start", () => {
    wrapper.find(".updateCompany_admin_input");
    expect(wrapper.state().selectedFile).toBe(null);
  });

  it("checks for on upload function to be called", () => {
    const e = { preventDefault: () => {} };
    wrapper.find(".updateCompany_admin_button").simulate("click", e);
  });
  it("checks for select file tab click ", () => {
    const e = { preventDefault: () => {} };
    wrapper.find(".updateCompany_admin_input").simulate("click", e);
  });
  it("should check for presense of onClickHandler function function", () => {
    let onClickHandler = jest.spyOn(wrapper.instance(), "onClickHandler");
    expect(onClickHandler).toBeTruthy();
  });

  it("should check for presense of onChangeHandler function", () => {
    let onChangeHandler = jest.spyOn(wrapper.instance(), "onChangeHandler");
    expect(onChangeHandler).toBeTruthy();
  });
});
