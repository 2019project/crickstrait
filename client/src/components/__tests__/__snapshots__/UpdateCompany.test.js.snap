// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar Component should mount the component 1`] = `
<div
  id="updateCompanyMainDiv"
>
  <a
    className="admin_close"
    href="#"
  >
    Ã—
  </a>
  <p
    id="updateCompanyMainDivText1"
  >
    Select a File
  </p>
  <div
    id="updateCompany_tooltip_div"
  >
    <i
      class="fa fa-info-circle"
    />
    <span
      id="updateCompany_tooltip_text"
    >
      <ul
        className="updateCompany_tooltip_text_list"
      >
        <li>
          File must be a 
          <b>
            csv
          </b>
           file.
        </li>
        <li>
          <b>
            Indicator Names
          </b>
           must be the same as used in the Application.
        </li>
        <li>
          Indicator Names on 
          <b>
            left
          </b>
           and their Data on
           
          <b>
            right
          </b>
        </li>
        <li>
          Date for the data must be in ISO format
          <b>
            ("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
          </b>
          .
        </li>
        <li>
          It should follow the below pattern.
        </li>
      </ul>
      <img
        alt="Sample Table"
        className="updateCompany_table_image"
        src="SampleData.png"
      />
    </span>
  </div>
  <input
    className="updateCompany_admin_input"
    name="file"
    onChange={[Function]}
    style={
      Object {
        "padding": "12px 35px 0px 35px",
      }
    }
    type="file"
  />
  <button
    className="btn btn-success btn-block updateCompany_admin_button"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-upload"
      style={
        Object {
          "marginRight": "5px",
        }
      }
    />
    Upload
  </button>
</div>
`;
